### DISCUSSION TITLE
Does gocron provide a lock between two jobs?

### DISCUSSION THREAD

**Discussion by Chen840302** [go-co-op/gocron#200] [Q&A]
For example, If I has two jobs like following:

scheduler.every(5).seconds().Do(funcA)
scheduler.every(6).seconds().Do(funcA)

They will execute at the same time at the 30 second,
Is there a way to execute only one of them?

**Comment by JohnRoesler** [comment]
With separate jobs - no. There is currently no locking mechanism supported. 

There is a feature request to support locking multiple instances of the same job https://github.com/go-co-op/gocron/issues/181

However, it looks to me like you're asking about locking based on the actual function being the same rather than the job. The library doesn't keep track of what the functions are beyond that it's an interface and the name of the function so it can be called. Equality isn't established at this point.

I'm curious, what's the usefulness of this locking ability?
